File: yandex_back_prep_python/leetcode/268. Missing Number.py
#O(n) spase O(1) time
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        
        h = {}

        for n in nums:
            if n not in h:
                h[n] = 1
            else:
                h[n] += 1
        
        for i in range(len(nums)+1):
            if i not in h:
                return i
            
        return True

File: yandex_back_prep_cpp/268. Missing Number.cpp
class Solution
{
public:
    int missingNumber(vector<int> &nums)
    {
        std::map<int, int> counter;
        int num_size = nums.size() + 1;

        for (int n : nums)
        {
            if (!counter.contains(n))
            {
                counter[n] = 1;
            }
            else
            {
                counter[n] = counter[n] + 1;
            }
        }

        for (int i = 0; i < num_size; i++)
        {
            if (!counter.contains(i))
            {
                return i;
            }
        }

        return 0;
    }
};